pkg/cmd/browse/browse.go

```diff
@@ -75,91 +75,90 @@ func NewCmdBrowse(f *cmdutil.Factory) *cobra.Command {
 			`),
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-
-			if len(args) > 1 {
-				opts.AdditionalArg = args[1]
+			opts.FlagAmount = cmd.Flags().NFlag()
+			if opts.FlagAmount > 1 {
+				return &cmdutil.FlagError{Err: fmt.Errorf("accepts 1 flag, %d flag(s) were recieved", opts.FlagAmount)}
 			}
-
+			// TODO test whether you can set cobra command arg range to 1 with a branch accepting an arg1
 			if len(args) > 0 {
 				opts.SelectorArg = args[0]
 			}
-			return openInBrowser(cmd, opts)
+
+			return runBrowse(opts)
 		},
 	}
 	cmdutil.EnableRepoOverride(cmd, f)
 	cmd.Flags().BoolVarP(&opts.ProjectsFlag, "projects", "p", false, "Open repository projects")
 	cmd.Flags().BoolVarP(&opts.WikiFlag, "wiki", "w", false, "Open repository wiki")
 	cmd.Flags().BoolVarP(&opts.SettingsFlag, "settings", "s", false, "Open repository settings")
-	cmd.Flags().BoolVarP(&opts.BranchFlag, "branch", "b", false, "Select another branch by passing in the branch name")
+	cmd.Flags().StringVarP(&opts.Branch, "branch", "b", "", "Select another branch by passing in the branch name")
 
 	return cmd
 }
 
-func openInBrowser(cmd *cobra.Command, opts *BrowseOptions) error {
+func runBrowse(opts *BrowseOptions) error {
 	help := "Use 'gh browse --help' for more information about browse\n"
+
 	baseRepo, err := opts.BaseRepo()
 	if err != nil {
 		return fmt.Errorf("unable to determine base repository: %w\n%s", err, help)
 	}
 
 	httpClient, _ := opts.HttpClient()
-	apiClient := api.NewClientFromHTTP(httpClient)
-	branchName, _ := api.RepoDefaultBranch(apiClient, baseRepo)
+	if err != nil {
+		return fmt.Errorf("unable to create an http client: %w\n%s", err, help)
+	}
 
-	if getFlagAmount(cmd) > 1 {
-		return fmt.Errorf("accepts 1 flag, %d flag(s) were recieved\n%s", getFlagAmount(cmd), help)
+	apiClient := api.NewClientFromHTTP(httpClient)
+	branchName, err := api.RepoDefaultBranch(apiClient, baseRepo)
+	if err != nil {
+		return fmt.Errorf("unable to determine default branch for %s: %w", ghrepo.FullName(baseRepo), err)
```

Comment:
This statement here always returns an `error` when running tests. I suspect it has something to do with using `httpmock` as our `HttpClient`? If we comment this statement out the tests we currently have work fine. Is there a way to "trick" the api into thinking it's getting a branch name? Here is one of the outputs of the test if we leave that statement in: 

`expected: *fmt.wrapError(&fmt.wrapError{msg:"unable to determine default branch for husrav/cli: Post \"https://api.github.com/graphql\": no registered stubs matched &{POST https://api.github.com/graphql HTTP/1.1 1 1 map[Content-Type:[application/json; charset=utf-8]] {{\"query\":\"\\n\\tfragment repo on Repository {\\n\\t\\tid\\n\\t\\tname\\n\\t\\towner { login }\\n\\t\\thasIssuesEnabled\\n\\t\\tdescription\\n\\t\\thasWikiEnabled\\n\\t\\tviewerPermission\\n\\t\\tdefaultBranchRef {\\n\\t\\t\\tname\\n\\t\\t}\\n\\t}\\n\\n\\tquery RepositoryInfo($owner: String!, $name: String!) {\\n\\t\\trepository(owner: $owner, name: $name) {\\n\\t\\t\\t...repo\\n\\t\\t\\tparent {\\n\\t\\t\\t\\t...repo\\n\\t\\t\\t}\\n\\t\\t\\tmergeCommitAllowed\\n\\t\\t\\trebaseMergeAllowed\\n\\t\\t\\tsquashMergeAllowed\\n\\t\\t}\\n\\t}\",\"variables\":{\"name\":\"cli\",\"owner\":\"husrav\"}}} 0x6a66c0 511 [] false api.github.com map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc0000200f8}", err:(*url.Error)(0xc00008f1d0)})`

Thanks in advance!

Reply:
The `httpmock.Registry` object is a fake implementation of HTTP that will block any API request that's not specifically stubbed, including this one. This error is to be expected if you didn't explicitly stub any responses.

You can stub the response for getting the default branch of a repository like so: https://github.com/cli/cli/blob/75abeb13a836cbf3faf43a2af12a8f1206787b4d/pkg/cmd/repo/view/view_test.go#L121-L122